src/ucp/core/ucp_am.c

```diff
@@ -62,17 +62,36 @@ void ucp_am_ep_init(ucp_ep_h ep)
 void ucp_am_ep_cleanup(ucp_ep_h ep)
 {
     ucp_ep_ext_proto_t *ep_ext = ucp_ep_ext_proto(ep);
+    ucp_recv_desc_t *first_rdesc, *tmp, *mid_rdesc;
```

Comment:
can have only:
```
ucp_recv_desc_t *rdesc, *tmp_rdesc;
```
and use `rdesc` and `tmp_rdesc` for first/middle fragments processing

Reply:
fixed